get-resource.sh

```diff
@@ -44,17 +45,27 @@ else
       unxz "$RHCOS_IMAGE_FILENAME_RAW"
     fi
 
-    qemu-img convert -O qcow2 -c "$RHCOS_IMAGE_FILENAME_OPENSTACK" "$RHCOS_IMAGE_FILENAME_COMPRESSED"
-    md5sum "$RHCOS_IMAGE_FILENAME_COMPRESSED" | cut -f 1 -d " " > "$RHCOS_IMAGE_FILENAME_COMPRESSED.md5sum"
+    qemu-img convert -O qcow2 -c "$RHCOS_IMAGE_FILENAME_QCOW" "$RHCOS_IMAGE_FILENAME_CACHED"
+    md5sum "$RHCOS_IMAGE_FILENAME_CACHED" | cut -f 1 -d " " > "$RHCOS_IMAGE_FILENAME_CACHED.md5sum"
 fi
 
-if [ -s "${RHCOS_IMAGE_FILENAME_COMPRESSED}.md5sum" ] ; then
+if [ -s "${RHCOS_IMAGE_FILENAME_CACHED}.md5sum" ] ; then
     cd /shared/html/images
     chmod 755 $TMPDIR
-    mv $TMPDIR $RHCOS_IMAGE_FILENAME_OPENSTACK
-    ln -sf "$RHCOS_IMAGE_FILENAME_OPENSTACK/$RHCOS_IMAGE_FILENAME_COMPRESSED" $FFILENAME
-    ln -sf "$RHCOS_IMAGE_FILENAME_OPENSTACK/$RHCOS_IMAGE_FILENAME_COMPRESSED.md5sum" "$FFILENAME.md5sum"
+    mv $TMPDIR $RHCOS_IMAGE_FILENAME_QCOW
+    ln -sf "$RHCOS_IMAGE_FILENAME_QCOW/$RHCOS_IMAGE_FILENAME_CACHED" $FFILENAME
+    ln -sf "$RHCOS_IMAGE_FILENAME_QCOW/$RHCOS_IMAGE_FILENAME_CACHED.md5sum" "$FFILENAME.md5sum"
 else
     rm -rf $TMPDIR
 fi
 
+# For backwards compatibility, if the rhcos image name contains -openstack, we want to
+# create a symlink to the original in the old format.  The old format had a substitution
+# which required the existence of -openstack in the name.
+if [[ ${RHCOS_IMAGE_FILENAME_QCOW} == *"-openstack"* ]] ; then
+    cd "/shared/html/images/${RHCOS_IMAGE_FILENAME_QCOW}"
+    RHCOS_IMAGE_FILENAME_COMPRESSED=${RHCOS_IMAGE_FILENAME_QCOW/-openstack/-compressed}
+    ln -sf "$RHCOS_IMAGE_FILENAME_CACHED" "$RHCOS_IMAGE_FILENAME_COMPRESSED"
+    ln -sf "$RHCOS_IMAGE_FILENAME_CACHED.md5sum" "$RHCOS_IMAGE_FILENAME_COMPRESSED.md5sum"
+fi
```

Comment:
If we did this earlier in the script, we could check for the existence of the old filename, and if it exists just rename it instead of downloading the same file again?

Reply:
Well we certainly could.  What are the chances it would be the same file?  This is mostly going to be on an upgrade.  My only real hesitation here is that I think it would be hard to test and it's a corner case optimization that adds one time value with more logic.

Reply:
Actually it might not be so hard to test.  Let me know and I can implement it, I just like things to stay simple. :)

Reply:
Currently there's no (automated) way to update the bootimage on upgrade but you're right that I expect it would change on a major version upgrade, so perhaps it is a corner case but it would avoid re-downloading the old image in the case where a user forgets to update the image prior to the upgrade :)

Personally I think it's worth doing as downloads from non-local mirrors can be really slow, but I'm not blocking on it :)